nixos-configuration-on-vm.md: Show how to run headless (#768)
NixOS/nix.dev: source/tutorials/nixos/nixos-configuration-on-vm.md

    run-nixos-vm
```

> To run the virtual machine in a new window (requires GUI environment):

Reviewer: ```suggestion
Run the virtual machine in the current terminal:
```

Let's not show all the options here, that's what reference documentation is for. The purpose of this tutorial (as any tutorial here) is to serve as a first encounter, with pointers to all the details.

I prefer the terminal version as that is or at least appears more lightweight, and showing a the flag here should provide enough of a hint that omitting it may give you a GUI.

```shell-session
./result/bin/run-nixos-vm
```

> To run it in the current terminal instead, pass `-nographic`:

Reviewer: Generally, instead of presenting multiple options - which is distracting to first-time users because they'll be busy with following *one* set of instructions, and also doesn't scale well, since where do you draw the line? – we should link to the manual right afterward, such as:

```
Refer to the Nixpkgs manual for a [list of all commmand-line options to `run-nixos-vm`](<URL>).
```

(If that exists. Otherwise there's likely a `--help` flag?)

```shell-session
./result/bin/run-nixos-vm -nographic
```

> You will see the boot process of the virtual machine and end at the `gdm` login screen where you can log in as `alice` with the password `testpw`.

Reviewer: I strongly recommend against describing perceptions, as not everyone can even have them. Describe what the machine does instead.

```suggestion
This will show the boot process of the virtual machine and end at the `gdm` login screen where you can log in as `alice` with the password `testpw`.
```

I currently can't check it myself: will that indeed be a terminal prompt? Isn't GDM a graphical tool?
